elyra/pipeline/handlers.py

```diff
@@ -86,10 +88,18 @@ async def post(self, *args, **kwargs):
         pipeline_definition = self.get_json_body()
         self.log.debug("JSON payload: %s", pipeline_definition)
 
-        pipeline = PipelineParser().parse(pipeline_definition)
+        root_dir = self.settings['server_root_dir']
+        expanded_path = get_expanded_path(root_dir)
+        validation_response = PipelineValidationManager().validate(pipeline=pipeline_definition,
+                                                                   root_dir=expanded_path)
+
+        validation_msg = validation_response.to_json()
+        json_msg = json.dumps(validation_response.to_json())
 
-        response = await PipelineProcessorManager.instance().process(pipeline)
-        json_msg = json.dumps(response.to_json())
+        if "issues" not in validation_msg:
+            pipeline = PipelineParser().parse(pipeline_definition)
+            response = await PipelineProcessorManager.instance().process(pipeline)
+            json_msg = json.dumps(response.to_json())
```

Comment:
As noted above, this indentation could be removed so that `response` either reflects a fatal validation response or the response from the operation.